src/sql-parser/src/ast/defs/query.rs

```diff
@@ -338,8 +344,11 @@ impl<T: AstInfo> CteBlock<T> {
                     names.push(&cte.alias.name);
                 }
             }
-            CteBlock::MutuallyRecursive(list) => {
-                for cte in list.iter() {
+            CteBlock::MutuallyRecursive(MutRecBlock {
```

Comment:
As an aside (don't need to do anything in this PR): having logic in the parser is unfortunate and it'd be nice for it to not be here.